### PR TITLE
Automate basic vsan test cases.

### LINKED ISSUE #1377: Automate VSAN policy test - basic
Please refer to #957 Scenario 3.

### PR DESCRIPTION
**Pull request by lipingxue**
Fixed #1377 

The fix is implemented by extending the existing vsan test.

### CODE REVIEW

File: tests/e2e/vsan_test.go [hunk 1]
```diff
@@ -63,6 +64,10 @@ func (s *VsanTestSuite) SetUpTest(c *C) {
 }
 
 func (s *VsanTestSuite) TearDownTest(c *C) {
+	if len(s.containerList) > 0 {
+		out, err := dockercli.RemoveAllContainers(s.config.DockerHosts[0])
```
Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: This is scary and stop us moving to run test in parallel. It is better to remove the container created by this test.

File: tests/e2e/vsan_test.go [hunk 2]
```diff
@@ -77,12 +82,24 @@ func (s *VsanTestSuite) TearDownTest(c *C) {
 
 var _ = Suite(&VsanTestSuite{})
 
+func (s *VsanTestSuite) newCName() string {
```
Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: method doc or improved method name would be useful to know about the intention behind this helper.

File: tests/e2e/vsan_test.go [hunk 3]
```diff
@@ -77,12 +82,24 @@ func (s *VsanTestSuite) TearDownTest(c *C) {
...
+	s.containerList = append(s.containerList, cname)
+	return cname
+}
+
 // Steps:
 // 1. Create a valid vsan policy
 // 2. Volume creation with valid policy should pass
 // 3. Valid volume should be accessible
+// 4. Start a container to mount the volume, and write message to the volume
+// 5. Start another container to mount the volume, and read message from volume and verify the
```
Comment (shaominchen) [inline tests/e2e/vsan_test.go]: I'm confused here: I don't see the test logic related to this step - start *another* container to mount the volume. Can you please clarify?

Comment (lipingxue) [inline tests/e2e/vsan_test.go]: I mean function ```dockercli.WriteToVolume``` internally will create a container and mount the volume.

Comment (shaominchen) [inline tests/e2e/vsan_test.go]: I still don't follow: the test logic shows that we are starting a container to mount the volume and write data to the volume, which is step 4. But where is the logic of "**start another container to mount the volume**" as described in step 5?

I guess step 5 should be just: "Read message from volume and verify the message..."

Comment (shaominchen) [inline tests/e2e/vsan_test.go]: As discussed offline, we are not *attaching* the volume to a different VM here, but just to *mounting* it to a different container *in the same VM*. That makes sense. Sorry for the confusion.

File: tests/e2e/vsan_test.go (hunk 1 above)
Comment (shaominchen) [inline tests/e2e/vsan_test.go]: Looks like we are already using this in volumeproperty_test.go.

Comment (govint) [inline tests/e2e/vsan_test.go]: Running test groups in parallel most likely will simply not work, given that all of these are operating off the default vmgroup and some are doing things like removing access privileges for the default vmgroup.

File: tests/e2e/vsan_test.go (hunk 2 above)
Comment (lipingxue) [inline tests/e2e/vsan_test.go]: Done

File: tests/e2e/vsan_test.go (hunk 1 above)
Comment (lipingxue) [inline tests/e2e/vsan_test.go]: OK. Will keep it as it is.

Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: Running test parallel is an action item for the 0.16 sprint to achieve "Overall time taken by each test run will be ~30minutes." ... we should such `dockercli.RemoveAllContainers` that blocks us moving test run in parallel. 

I don't want to repeat again ... test container is a test item and its test responsibility to remove.

> Running test groups in parallel most likely will simply not work,

Which ever the test is not possible they will be identified and we may put into serial run bucket. 

> Looks like we are already using this in volumeproperty_test.go.

It was a miss from the review and by taking account of above example we should not use this approach. I discussed with @ashahi1  to remove usage of `dockercli.RemoveAllContainers` and remove only the containers created by test. 

/CC @ashahi1

Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: @lipingxue By looking at your test code I don't see a need for cleaning up containers. Let's remove it.

File: tests/e2e/vsan_test.go [hunk 4]
```diff
@@ -77,12 +82,25 @@ func (s *VsanTestSuite) TearDownTest(c *C) {
 
 var _ = Suite(&VsanTestSuite{})
 
+// This function is used to return a unique container name used for "vsan_test"
+func (s *VsanTestSuite) newCName() string {
```
Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: this method can be removed and replaced by inline invocation.

Comment (lipingxue) [inline tests/e2e/vsan_test.go]: Code is more clean with this helper function and I will keep it as it is.

File: tests/e2e/vsan_test.go (hunk 1 above)
Comment (ashahi1) [inline tests/e2e/vsan_test.go]: I will be removing ```dockercli.RemoveAllContainers``` function and will only remove the containers that test created.

File: tests/e2e/vsan_test.go (hunk 4 above)
Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: There is not much left in the method other than one line `inputparams.GetUniqueContainerName("vsan_test")` after syncing up with master code. Containers will be removed automatically hence no required of L88 `s.containerList = append(s.containerList, cname)` ... This method left out to be as inline function.

File: tests/e2e/vsan_test.go [hunk 5]
```diff
@@ -41,9 +41,10 @@ import (
 type VsanTestSuite struct {
 	config *inputparams.TestConfig
 
-	vsanDSName string
-	volumeName string
-	policyList []string
+	vsanDSName    string
+	volumeName    string
+	policyList    []string
+	containerList []string
```
Comment (shuklanirdesh82) [inline tests/e2e/vsan_test.go]: minor: this should be removed.

**Pull request merged**